macro/src/generate.rs

```diff
@@ -19,6 +19,13 @@ pub(crate) fn generate(item: &BoundedInteger, tokens: &mut TokenStream) {
     generate_from_str(item, tokens);
     generate_fmt_traits(item, tokens);
     generate_to_primitive_traits(item, tokens);
+    if item.repr.is_usize() {
+        generate_index_traits(item, tokens);
+
+        if item.std {
```

Comment:
One thing I did after your last PR was merged was move these implementations [from the `std` to the `alloc` feature flag](https://github.com/Kestrer/bounded-integer/commit/4d42e407618a2c16d9156f0048c26ba6ede44ebf), as `Vec` and `VecDeque` are present in not just libstd but also liballoc, which the library between core and std that runs with a memory allocator but not necessarily an operating system.

So there’s a tricky thing you have to do here: Unlike `core` and `std`, we _can’t_ rely on `::alloc::vec::Vec` being present. Sometimes it is present, if the user has placed `extern crate alloc;` at their crate root, but most people don’t do that. Hence, what we have to do is pass both the `std` and `alloc` feature flags to the macro and change our generated code accordingly:

* If the user enables neither feature flag, do not generate anything;
* If the user enables `alloc` but not `std`, generate references to `::alloc::vec::Vec` and `::alloc::collections::VecDeque`;
* If the user enables `alloc` and `std`, generate references to `::std::vec::Vec` and `::std::collections::VecDeque`.